pkg/cluster/service_test.go

```diff
@@ -0,0 +1,216 @@
+package cluster_test
+
+import (
+	"github.com/codeready-toolchain/toolchain/pkg/cluster"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"testing"
+)
+
+func newKubeFedCluster(name, secName string, status v1beta1.KubeFedClusterStatus, labels map[string]string) (*v1beta1.KubeFedCluster, *corev1.Secret) {
+	logf.SetLogger(zap.Logger())
+	gock.New("http://cluster.com").
+		Get("api").
+		Persist().
+		Reply(200).
+		BodyString("{}")
+	secret := &corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      secName,
+			Namespace: "test-namespace",
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"token": []byte("mycooltoken"),
+		},
+	}
+
+	return &v1beta1.KubeFedCluster{
+		Spec: v1beta1.KubeFedClusterSpec{
+			SecretRef: v1beta1.LocalSecretReference{
+				Name: secName,
+			},
+			APIEndpoint: "http://cluster.com",
+			CABundle:    []byte{},
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: "test-namespace",
+			Labels:    labels,
+		},
+		Status: status,
+	}, secret
+}
+
+func TestAddKubeFedClusterAsMember(t *testing.T) {
+	// given
+	defer gock.OffAll()
```

Comment:
```suggestion
	defer gock.Off()
```
It's safer to use `gock.Off()` instead. Go runs tests in different packages in parallel by default. Killing all unmatched requests can cause race condition between tests from different packages and can cause test failures.